src/test/unit/engine/HelloRegionTest.cpp

```diff
@@ -98,21 +111,28 @@ TEST(HelloRegionTest, demo) {
     ss.seekg(0);
     net2.load(ss);
   }
-//  EXPECT_EQ(net, net2);
-	//std::cout << "Loading complete. \n";
+  EXPECT_EQ(net, net2) << "Restored network should be the same as original.";
 
-  std::shared_ptr<Region> region2 = net2.getRegion("region"); //TODO add more checks and asserts here
-  region2->executeCommand(loadFileArgs);
+  std::shared_ptr<Region> region2 = net2.getRegion("region");
   ArrayRef outputArray2 = region2->getOutputData("dataOut");
-  const Real64 *buffer2 = (const Real64 *)outputArray2.getBuffer();
 
-  net.run(DATA_SIZE);
-  net2.run(DATA_SIZE);
+  // fetch the data rows for both networks.
+  net.run((int)data_rows-1);
```

Comment:
why rows-1?

Reply:
> why rows-1?

Because the first row was already consumed before the save.